homeassistant/components/evohome/__init__.py

```diff
@@ -723,7 +725,7 @@ async def _update_schedule(self) -> None:
 
         assert isinstance(self._evo_device, evo.HotWater | evo.Zone)  # mypy check
 
-        self._schedule = await self._evo_broker.call_client_api(
+        self._schedule = await self._evo_broker.call_client_api(  # type: ignore[assignment]
```

Comment:
Do we need to declare `self._schedule` as optional?

Reply:
I saw that we had a check for falsy for `self._schedule` in another part of the file so I figured it could be an actual case or that it wouldn't matter to type it as optional if we already checked that.

We may be able to improve this by using some of the new typing tools, similar to how we type decorators nowadays. I think we can look at that in a follow up.

Let's keep it like this for now.

Reply:
Here's an example of what I was thinking:

https://github.com/home-assistant/core/blob/f0104d6851e8d9370091510c4101d5c0ea64f3cb/homeassistant/core.py#L604-L607

If the target function is typed, the type checker will understand that the target return value will be the return value of the wrapper too.

Reply:
Fair enough. I think that code is an anachronism:
```
if not self._schedule.get("DailySchedules"):
```
... would do.